cas-server-support-jdbc/src/main/java/org/jasig/cas/services/JpaServiceRegistryDaoImpl.java

```diff
@@ -35,6 +37,7 @@
     @PersistenceContext
     private EntityManager entityManager;
 
+    @Transactional(readOnly = false)
```

Comment:
I'm not sure about this either. It might also depend on the chosen isolation level.

As generally the first operation is a read and then a write, I expect no side effects, compared to a write followed by a read because in that case, the write would be visible only to the following read in the same transaction.

I remember some reported concerns / issues regarding performance and lock with the DB support, I guess having smaller transactions will help.
So I tend to think that your changes are a good choice.